backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java

```diff
@@ -225,13 +226,26 @@ public List<Facet> getBundleFacets(@Context SecurityContext sec, @QueryParam("in
     })
     @RolesAllowed(ConsoleIdentityProvider.RBAC_WRITE_NOTIFICATIONS)
     @Transactional
-    public CreateBehaviorGroupResponse createBehaviorGroup(@Context SecurityContext sec,
-                              @RequestBody(required = true) @Valid @NotNull CreateBehaviorGroupRequest request) {
+    public CreateBehaviorGroupResponse createBehaviorGroup(
+        @Context final SecurityContext sec,
+        @RequestBody(required = true) @Valid @NotNull final CreateBehaviorGroupRequest request
+    ) {
         String accountId = getAccountId(sec);
         String orgId = getOrgId(sec);
 
+        // Set the bundle ID from the incoming request. If the user provided a
+        // bundle name instead, fetch the bundle ID from the database. This is
+        // to make the frontend's life easier, in the way that they won't need
+        // to fetch the bundles to get the exact bundle ID. More information in
+        // the request's class, or in RHCLOUD-22513.
+        UUID bundleId = request.bundleId;
+        if (bundleId == null && !StringUtils.isBlank(request.bundleName)) {
```

Comment:
We already know from the `@AssertFalse` constraint that `bundleName` is not blank if `bundleId` is null.
```suggestion
        if (bundleId == null) {
```

Reply:
Great catch! Refactored.